pkg/mimir/modules.go

```diff
@@ -1050,6 +1051,10 @@ func (t *Mimir) initRuler() (serv services.Service, err error) {
 			return nil, fmt.Errorf("could not create queryable for ruler: %w", err)
 		}
 
+		// Wrap the queryable to ensure all queries have a memory tracker in the context.
+		// This is needed for ruler operations like alert state restoration that may bypass
+		// the query engine's NewInstantQuery/NewRangeQuery methods.
```

Comment:
Could you please expand the comment here to explain that the `Queryable` won't inject a new memory consumption tracker if there's already one on the context?